src/Sentry.Unity.Native/SentryNativeBridge.cs

```diff
@@ -42,9 +96,85 @@ public static void Init(SentryUnityOptions options)
         [DllImport("sentry")]
         private static extern void sentry_options_set_release(IntPtr options, string release);
 
-        // TODO we could set a logger for sentry-native, forwarding the logs to `options.DiagnosticLogger?`
-        // [DllImport("sentry")]
-        // private static extern void sentry_options_set_logger(IntPtr options, IntPtr logger, IntPtr userData);
+        [DllImport("sentry")]
+        private static extern void sentry_options_set_debug(IntPtr options, int debug);
+
+        [DllImport("sentry")]
+        private static extern void sentry_options_set_environment(IntPtr options, string environment);
+
+        [DllImport("sentry")]
+        private static extern void sentry_options_set_sample_rate(IntPtr options, double rate);
+
+        [DllImport("sentry")]
+        private static extern void sentry_options_set_database_path(IntPtr options, string path);
+
+        [DllImport("sentry")]
+        private static extern void sentry_options_set_database_pathw(IntPtr options, [MarshalAs(UnmanagedType.LPWStr)] string path);
+
+        [DllImport("sentry")]
+        private static extern void sentry_options_set_auto_session_tracking(IntPtr options, int debug);
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl, SetLastError = true)]
+        private delegate void sentry_logger_function_t(int level, string message, IntPtr argsAddress, IntPtr userData);
+
+        [DllImport("sentry")]
+        private static extern void sentry_options_set_logger(IntPtr options, sentry_logger_function_t logger, IntPtr userData);
+
+        // The logger we should forward native messages to.
+        private static IDiagnosticLogger? _logger;
+        private static Mutex _loggerMutex = new Mutex();
+
+        private static IDiagnosticLogger? _getLogger()
+        {
+            _loggerMutex.WaitOne();
+            var result = _logger;
+            _loggerMutex.ReleaseMutex();
+            return result;
+        }
+        private static void _setLogger(IDiagnosticLogger? logger)
+        {
+            _loggerMutex.WaitOne();
+            _logger = logger;
+            _loggerMutex.ReleaseMutex();
+        }
+
+        // This method is called from the C library
+        [MonoPInvokeCallback(typeof(sentry_logger_function_t))]
+        private static void nativeLog(int cLevel, string message, IntPtr args, IntPtr userData)
+        {
+            var logger = _getLogger();
+            if (logger is null)
+                return;
```

Comment:
nit: Please always use `{` in blocks
```suggestion
            if (logger is null)
            {
                return;
            }
```